cmd/profile/delete.go

```diff
@@ -24,7 +26,15 @@ func deleteProfileCommand() *cobra.Command {
 			if err != nil {
 				return fmt.Errorf("failed to get k8s client config: %s", err)
 			}
-			return deleteProfile(config, ns, args[0])
+			profileName := args[0]
+			err = deleteProfile(config, ns, profileName)
+			if err != nil {
+				fmt.Printf("%s not found, assuming legacy profile\n", profileName)
```

Comment:
You can't assume that a non-nil error indicates "not found". Another error could have occurred, e.g. a network failure, in which case you want to fail and return. There are utilities to check whether an error returned by a kubernetes client is in fact a NotFound. Let me know if you need help finding it. I believe some parts of arlon code already use it. If not, search inside ArgoCD repo. Look for something like IsNotFound().

Reply:
Thanks. I missed that.